src/timer/timer_game_common.h

```diff
@@ -0,0 +1,205 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file timer_game_common.h Definition of the common class inherited by both calendar and economy timers. */
+
+#ifndef TIMER_GAME_COMMON_H
+#define TIMER_GAME_COMMON_H
+
+#include "stdafx.h"
+#include "../core/strong_typedef_type.hpp"
+
+enum DaysTillMonth {
+	ACCUM_JAN = 0,
+	ACCUM_FEB = ACCUM_JAN + 31,
+	ACCUM_MAR = ACCUM_FEB + 29,
+	ACCUM_APR = ACCUM_MAR + 31,
+	ACCUM_MAY = ACCUM_APR + 30,
+	ACCUM_JUN = ACCUM_MAY + 31,
+	ACCUM_JUL = ACCUM_JUN + 30,
+	ACCUM_AUG = ACCUM_JUL + 31,
+	ACCUM_SEP = ACCUM_AUG + 31,
+	ACCUM_OCT = ACCUM_SEP + 30,
+	ACCUM_NOV = ACCUM_OCT + 31,
+	ACCUM_DEC = ACCUM_NOV + 30,
+};
+
+/** Number of days to pass from the first day in the year before reaching the first of a month. */
+static constexpr uint16_t _accum_days_for_month[] = {
+	ACCUM_JAN, ACCUM_FEB, ACCUM_MAR, ACCUM_APR,
+	ACCUM_MAY, ACCUM_JUN, ACCUM_JUL, ACCUM_AUG,
+	ACCUM_SEP, ACCUM_OCT, ACCUM_NOV, ACCUM_DEC,
+};
```

Comment:
These can now be moved into the .cpp. That was kinda the point of moving the function, not? :)